UndertaleModLib/Models/UndertaleRoom.cs

```diff
@@ -1415,12 +1481,15 @@ public int WrappedFrameIndex
             }
             public float Rotation { get; set; }
             public float OppositeRotation => 360F - Rotation;
-            public int XOffset => Sprite is not null
-                                  ? X - Sprite.OriginX + (Sprite.Textures.ElementAtOrDefault((int)FrameIndex)?.Texture?.TargetX ?? 0)
-                                  : X;
-            public int YOffset => Sprite is not null
-                                  ? Y - Sprite.OriginY + (Sprite.Textures.ElementAtOrDefault((int)FrameIndex)?.Texture?.TargetY ?? 0)
-                                  : Y;
+
+            public int SpriteXOffset => Sprite != null
+                                        ? (-1 * Sprite.OriginXWrapper) + (Sprite.Textures.ElementAtOrDefault(WrappedFrameIndex)?.Texture?.TargetX ?? 0)
+                                        : 0;
+            public int SpriteYOffset => Sprite != null
+                                        ? (-1 * Sprite.OriginYWrapper) + (Sprite.Textures.ElementAtOrDefault(WrappedFrameIndex)?.Texture?.TargetY ?? 0)
+                                        : 0;
+            public int XOffset => X + SpriteXOffset;
+            public int YOffset => Y + SpriteYOffset;
```

Comment:
Mind adding documentation for these?

Reply:
If I'll add that here, then I also should add descriptions to other properties as well.

Reply:
Not necessarily, but would be nicer, yeah. 
If you plan to do that, maybe do that in another PR?

Reply:
Yeah, probably will do that.